cloudstack.go

```diff
@@ -696,30 +711,46 @@ func (d *Driver) setDiskOffering(diskOffering string, diskOfferingID string) err
 }
 
 func (d *Driver) setNetwork(networkName string, networkID string) error {
-	d.Network = networkName
-	d.NetworkID = networkID
-
-	if d.Network == "" && d.NetworkID == "" {
+	if networkName == "" && networkID == "" {
+		d.NetworkID = nil
+		d.Network = nil
 		return nil
 	}
 
 	cs := d.getClient()
 	var network *cloudstack.Network
 	var err error
-	if d.NetworkID != "" {
-		network, _, err = cs.Network.GetNetworkByID(d.NetworkID, d.setParams)
+	var networkIDs, networkNames []string
+
+	if networkID != "" {
+		networkIDs = make([]string, len(strings.Split(networkID, ",")))
```

Comment:
You should call Split only once and reuse the result here and in the range statement.